fms2_io/netcdf_io.F90

```diff
@@ -767,6 +768,50 @@ subroutine append_compressed_dimension(fileobj, dim_name, npes_corner, &
   fileobj%compressed_dims(n)%nelems = sum(fileobj%compressed_dims(n)%npes_nelems)
 end subroutine append_compressed_dimension
 
+!> @brief Add a "compressed" unlimited dimension to a netcdf file.
+!! @note Here compressed means that every rank has a different dimension_length
+!compressed. This was written specifically for the icebergs restarts.
+subroutine register_unlimited_compressed_axis(fileobj, dimension_name, dimension_length)
+  class(FmsNetcdfFile_t), intent(inout) :: fileobj !< File object.
+  character(len=*), intent(in) :: dimension_name !< Dimension name.
+  integer, intent(in) :: dimension_length !< Dimension length for the current rank
+
+  integer                            :: dim_len    !< Dimension length
```

Comment:
Is this variable needed?
```fortran
  dim_len = dimension_length
```
it could just be a pointer in this case.

```fortran
dim_len = sum(npes_count)
```
I don't think it's used after this, so this is set for no reason.